internal/api/controllers/private/highlevelJobRequest.go

```diff
@@ -0,0 +1,109 @@
+package private
+
+import (
+	"fmt"
+	"net/http"
+	"playbook-dispatcher/internal/api/instrumentation"
+	"playbook-dispatcher/internal/api/middleware"
+	"playbook-dispatcher/internal/common/utils"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+)
+
+func (this *controllers) ApiInternalHighlevelJobRequest(ctx echo.Context) error {
+	var input RunInputV2List
+
+	err := utils.ReadRequestBody(ctx, &input)
+	if err != nil {
+		utils.GetLogFromEcho(ctx).Error(err)
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	for _, run := range input {
+		requestType, err := validateJobRequestFields(run)
+		if err != nil {
+			instrumentation.InvalidHighLevelJobRequest(ctx, requestType, err)
+			return invalidRequest(ctx, err)
+		}
+	}
+
+	// processing individual requests concurrently
+	result := input.PMapRunCreated(func(runInputV2 RunInputV2) *RunCreated {
+		requestTypeLabel := getRequestTypeLabel(runInputV2)
+		context := utils.WithOrgId(ctx.Request().Context(), string(runInputV2.OrgId))
+		context = utils.WithRequestType(context, requestTypeLabel)
+
+		recipient := parseValidatedUUID(string(runInputV2.Recipient))
+
+		hosts := parseRunHosts(runInputV2.Hosts)
+
+		var parsedSatID *uuid.UUID
+		if runInputV2.RecipientConfig != nil && runInputV2.RecipientConfig.SatId != nil {
+			parsedSatID = utils.UUIDRef(parseValidatedUUID(string(*runInputV2.RecipientConfig.SatId)))
+		}
+
+		runInput := RunInputV2GenericMap(runInputV2, recipient, hosts, parsedSatID, this.config)
+
+		hostConnectorDetails, err := this.inventoryConnectorClient.GetHostConnectionDetails(
+			context,
+			extractInventoryIds(runInput.Hosts),
+			this.config.GetString("inventory.connector.ordered.how"),
+			this.config.GetString("inventory.connector.ordered.by"),
+			this.config.GetInt("inventory.connector.limit"),
+			this.config.GetInt("inventory.connector.offset"),
+		)
+		if err != nil {
+			utils.GetLogFromContext(context).Error(err)
+			return handleRunCreateError(err)
+		}
+
+		satellite, directConnected, noRhc := sortHostsByRecipient(hostConnectorDetails)
```

Comment:
These are the recipient ids for the inventory host ids that were passed in.  I think we are basically ignoring these.